### DISCUSSION TITLE
Using chartjs-chart-geo in widgets

### DISCUSSION THREAD

**Discussion by sasaorasanin** [filamentphp/filament#11472] [Help]
### Package

Widgets

### Package Version

v3.2.33

### How can we help you?

How can we use chart.js  plugins inside filament windets, like chartjs-chart-geo plugin?
Or how to register specific controller?
Thanks

**Comment by leandrocfe** [comment]
Did you try to [register the JS file](https://filamentphp.com/docs/3.x/support/assets#registering-javascript-files)?

**Comment by sandermanneke** [comment]
I have a similar issue: I am trying to add https://chartjs-plugin-datalabels.netlify.app/guide/getting-started.html#installation plugin to Chart.js

Searching the discussions I found:

this https://github.com/filamentphp/filament/discussions/6088 but could not get that to work for me. Also it seems that it overrides Chart in Alpine and will break future changes in Filament to Chart.

this introduces a new blade view for charts which has the same problem for me with regards to maintenance: https://github.com/filamentphp/filament/discussions/4155

On this PR how to use an adapter https://github.com/filamentphp/filament/pull/9514 @danharrin just includes it in the codebase. That gives me the impression there is no elegant way to manage plugins and adapters.

How should I go about this?

**Reply by danharrin** [reply]
There isnt a way to do this from userland atm, if you want to work out how we should implement it go ahead
